worlds/clique/Rules.py

```diff
@@ -0,0 +1,8 @@
+from BaseClasses import CollectionState
+
+
+def has_unlocked_button(state: CollectionState, player: int) -> bool:
+    if getattr(state.multiworld, "hard_mode")[player]:
+        return state.has("Button Activation", player)
```

Comment:
Assigning different callables based on difficulty (rather than checking difficulty in the callable) would have way better performance.
```suggestion
    if getattr(state.multiworld, "hard_mode")[player]:
        return state.has("Button Activation", player)
```

Reply:
Made a change that might work out better.